pkg/reader/ccip.go

```diff
@@ -550,45 +550,113 @@ func (r *ccipChainReader) GetChainFeePriceUpdate(ctx context.Context, selectors
 	return feeUpdates
 }
 
-func (r *ccipChainReader) DiscoverContracts(
+func (r *ccipChainReader) discoverDestinationContracts(
 	ctx context.Context,
-	destChain cciptypes.ChainSelector,
+	allChains []cciptypes.ChainSelector,
 ) (ContractAddresses, error) {
-	if err := validateExtendedReaderExistence(r.contractReaders, destChain); err != nil {
-		return nil, err
+	// Exit without an error if we cannot read the destination.
```

Comment:
Moved this to `DiscoverContracts` along with a comment. The handling is a little different than elsewhere because it's important to continue after this error in order to handle cases where a node is configured such that they can read from a source chain but not the dest chain.